Added locale to redirect_to helper. Made use of it in the calls in re…

JVickery-TBS opened the pull request:
…quest_reset.

Fixes #

Password reset form redirect does not maintain current locale. It just redirects to "/". Added locale to helper method and made use of it in request_reset redirects. This is just a small hot fix which will most likely e removed soon from the upgrade to ckan2.9

### Proposed fixes:



### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport
- [X] None of the above

Please [X] all the boxes above that apply

wardi commented on ckan/controllers/user.py:
```diff
@@ -512,14 +512,13 @@ def request_reset(self):
                         _(u'Error sending the email. Try again later '
                           'or contact an administrator for help')
                     )
-                    log.exception(e)
```
better keep the log here

wardi commented on ckan/lib/helpers.py:
```diff
@@ -189,8 +189,12 @@ def redirect_to(*args, **kw):
         _url = url_for(*uargs, **kw)
 
+    locale = kw.pop('locale', '')
+    if len(locale) > 0:
```
```suggestion
    if locale:
```

wardi commented:
looks like this is also an issue upstream (there's a `redirect_to('/user/reset')` in views/user.py and the helper code looks the same) also as a really minor point this change isn't handling root_path correctly.

Would you submit a similar fix to core instead so it can be discussed and assigned?

JVickery-TBS commented:
@wardi okay this code should be good now. I added your suggestions (also oops I removed that log statement).

I also added the locale to that other `redirect_to('/user/reset')`

I will work on a PR for ckan/ckan master

wardi commented:
looks good. Could the same be submitted upstream?
